server/api/events/events.controller.js

```diff
@@ -124,41 +115,41 @@ export const indexByUser = (req, res) => {
     .populate('participants.userId', 'avatar emails name')
     .exec()
     .then((events) => {
-      const nEvents = _.clone(events);
       events.forEach((event, index) => {
         event.participants.forEach((participant, indexParticipant) => {
           // if he is the owner then dont show the event.
-          if (participant.status === 0 && participant.userId._id.toString() === req.user._id.toString()) {
-            nEvents.splice(index, 1);
-          } else if (participant.status === 0 && participant.userId._id.toString() !== req.user._id.toString()) {
-            nEvents[index].participants.splice(indexParticipant, 1);
+          if (participant.status === 0 &&
+              participant.userId._id.toString() === req.user._id.toString()) {
+            events[index] = null;
+          } else
+          if (participant.status === 0 &&
+              participant.userId._id.toString() !== req.user._id.toString()) {
+            events[index].participants[indexParticipant] = null;
           }
         });
       });
-      return nEvents;
+
```

Comment:
A filter here I deemed not to be practical.  There was a bug in this code that I fixed with the changes to the code above and the new code below.